### PR TITLE
fix(Bonus Pagamenti Digitali): [#175923444] Bottom sheet should be closed on hardware back button press

### PR DESCRIPTION
**Pull request by Undermaken**
## Short description
This PR makes the bottom sheet disposable when hardware back button is pressed on Android devices.

### changes
- refactoring: now all bottom sheets are created starting from internal function factories
- add new bottom sheet function factory `useIOBottomSheetRaw`: useful when you need to handle the BS dismiss inside your inner component or when you need to change BS title basing from specific condition

https://user-images.githubusercontent.com/822471/102590794-57e7f180-4111-11eb-9830-daefd0ae0d94.mov

### CODE REVIEW

File: ts/features/wallet/component/RemovePaymentMethod.tsx [hunk 1]
```diff
@@ -55,10 +54,10 @@ export const RemovePaymentMethod: React.FunctionComponent<Props> = props => (
...
   representation: PaymentMethodRepresentation
 ) => {
-  const { present, dismiss } = useBottomSheetModal();
+  const { present, dismiss } = useIOBottomSheetRaw(350);
 
-  const openModalBox = (onConfirm: () => void) => {
-    const bottomSheetProps = bottomSheetRawConfig(
+  const openBottomSheet = (onConfirm: () => void) =>
+    present(
       <RemovePaymentMethod
```
Comment (debiff) [inline ts/features/wallet/component/RemovePaymentMethod.tsx:61]: Why the `RemovePaymentMethod` component is exported?

**Review by debiff — APPROVED** [review]

File: ts/features/wallet/component/RemovePaymentMethod.tsx (hunk 1 above)
Comment (Undermaken) [reply ts/features/wallet/component/RemovePaymentMethod.tsx:61]: I can't get it. What do you mean with `exported`?

Comment (debiff) [reply ts/features/wallet/component/RemovePaymentMethod.tsx:61]: [Here](https://github.com/pagopa/io-app/blob/50b3ffccdc301da4beef4462881aa5eea10531d2/ts/features/wallet/component/RemovePaymentMethod.tsx#L26).
This component is only used inside this file, is `export` key needed?

File: ts/components/wallet/WalletHomeHeader.tsx [hunk 2]
```diff
@@ -92,50 +70,51 @@ const WalletHomeHeader: React.FC<Props> = (props: Props) => {
...
-            )}
-          </>
-        )}
-      />,
-      I18n.t("global.buttons.add"),
-      315,
-      closeBS
-    );
+  const closeBS = () => {
+    dismiss();
```
Comment (CrisTofani) [inline ts/components/wallet/WalletHomeHeader.tsx:113]: Is this needed to wrap a single instruction?

We may use disimiss call directly inside this code block:
```javascript
            <ButtonDefaultOpacity
              onPress={() => {
                dismiss();
                item.onPress();
              }}
              style={styles.container}
              onPressWithGestureHandler={true}
            >
```

what do you think?

**Review by CrisTofani — CHANGES_REQUESTED** [review]

File: ts/features/wallet/component/RemovePaymentMethod.tsx (hunk 1 above)
Comment (Undermaken) [reply ts/features/wallet/component/RemovePaymentMethod.tsx:61]: Got it. Thanks
https://github.com/pagopa/io-app/pull/2614/commits/0a9d2ede20acc3d97c4fc05f6ee94b5736c069a0

File: ts/components/wallet/WalletHomeHeader.tsx (hunk 2 above)
Comment (Undermaken) [reply ts/components/wallet/WalletHomeHeader.tsx]: sure! thx

https://github.com/pagopa/io-app/pull/2614/commits/2ca2f86c3926ecf4602d6bc2061355b3700d2ba4

**Review by CrisTofani — APPROVED** [review]

**Pull request merged**